includes/class-amp-theme-support.php

```diff
@@ -1635,13 +1641,29 @@ public static function ensure_required_markup( DOMDocument $dom, $script_handles
 		}
 		$head->insertBefore( $meta_viewport, $meta_charset->nextSibling );
 
+		// Merge meta amp-script-src elements.
+		$first_meta_amp_script_src = array_shift( $meta_amp_script_srcs );
+		foreach ( $meta_amp_script_srcs as $meta_amp_script_src ) {
+			$first_meta_amp_script_src->setAttribute(
```

Comment:
Seems wasteful to constantly rewrite this attribute (although I ignore how much processing happens in the DOM abtraction). Would make more sense to collect and merge as a variable, and then write (i.e. `setAttribute()`) only once.

Also, storing the first element, and then iterating over the rest is a rather arbitrary logic. Using `array_reduce()` would be conceptually clearer.

Reply:
Yeah, I think calling `setAttribute` multiple times may not be a concern like it would be in the browser, since there are no layout calculations being performed. Nevertheless, minimizing writes and redundant string concatenations seems beneficial. To that end, I think actually `array_reduce()` isn't the best fit here because it would do more concatenation operations than required, while all we need to do is push onto a single copy of an array. At the same time, we also need to remove the elements from the DOM, and that doesn't seem super clean to do in an `array_reduce()`.

Please take a look at 03d587ab.